components/quickbooks/actions/sparse-update-invoice/sparse-update-invoice.mjs

```diff
@@ -0,0 +1,98 @@
+import { ConfigurationError } from "@pipedream/platform";
+import quickbooks from "../../quickbooks.app.mjs";
+
+export default {
+  key: "quickbooks-sparse-update-invoice",
+  name: "Sparse Update Invoice",
+  description: "Sparse updating provides the ability to update a subset of properties for a given object; only elements specified in the request are updated. Missing elements are left untouched. The ID of the object to update is specified in the request body.​ [See docs here](https://developer.intuit.com/app/developer/qbo/docs/api/accounting/all-entities/invoice#sparse-update-an-invoice)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    quickbooks,
+    invoiceId: {
+      propDefinition: [
+        quickbooks,
+        "invoiceId",
+      ],
+    },
+    lineItems: {
+      propDefinition: [
+        quickbooks,
+        "lineItems",
+      ],
+    },
+    customerRefValue: {
+      label: "Customer Reference Value",
+      type: "string",
+      description: "Reference to a customer or job. Query the Customer name list resource to determine the appropriate Customer object for this reference. Use `Customer.Id` from that object for `CustomerRef.value`.",
+    },
+    customerRefName: {
+      propDefinition: [
+        quickbooks,
+        "customerRefName",
+      ],
+    },
+    currencyRefValue: {
+      propDefinition: [
+        quickbooks,
+        "currencyRefValue",
+      ],
+    },
+    currencyRefName: {
+      propDefinition: [
+        quickbooks,
+        "currencyRefName",
+      ],
+    },
```

Comment:
Shouldn't all theses properties have `async options()` in order to query those objects and select the one the user needs to?

Reply:
I didn't do this to keep the same pattern as the other actions

Reply:
I'm suggesting this because alway it's a good practice to give the user a good experience. In this case they would have the oportunity to just select an item from a dropbox using Pipedream's UI. So in this case we can prioritize UX over code implementation right?

Reply:
I agree with you, but I think for UX to be good, I'm going to need to improve these props in every action that uses this, otherwise I think it's going to be confusing. What do you think?

Reply:
@luancazarine I think that would be an excellent idea. I guess you can do it in a different PR that you can create in that way we can move on with this one right?
Then you can create an improvement PR on your own. And still you can limit to fix only `sparse-update-invoice.mjs` this time to make it as a template for the future PR.